packages/data/src/factory.js

```diff
@@ -39,11 +39,15 @@
  * @return {Function} Registry selector that can be registered with a store.
  */
 export function createRegistrySelector( registrySelector ) {
+	let selector;
 	// Create a selector function that is bound to the registry referenced by `selector.registry`
 	// and that has the same API as a regular selector. Binding it in such a way makes it
 	// possible to call the selector directly from another selector.
-	const selector = ( ...args ) =>
-		registrySelector( selector.registry.select )( ...args );
+	const wrappedSelector = ( ...args ) => {
+		selector =
+			selector || registrySelector( wrappedSelector.registry.select );
```

Comment:
Maybe it's faster to do `if ( ! selector ) { selector = ... }`? That avoids a variable assignment on every call. But I don't know, it looks like an easy thing for the engine to optimize out.